### PR TITLE
API-33700 hourly submission error report to slack

### PR DESCRIPTION
**Pull request by rockwellwindsor-va**
## Summary
- Updates periodic jobs table
- Adds sidekiq job to handle hourly run
- Adds service to format message and send slcak message
- Adds RSpec tests for newly created sidekiq job and service class
- *This work is behind a feature toggle (flipper): YES/NO* - NO
- *(Which team do you work for, does your team own the maintenance of this component?)* - Dash, it appears we do not own the periodic jobs file.

## Related issue(s)
[API-33700](https://jira.devops.va.gov/browse/API-33700)

## Testing done
- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change* - v Did not notify on the hour
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:* - NO

#### Testing notes
* A sample of the test message can be seen below in the "screenshots" (This will be run in Production so the ENV values would say production, but to test it was run in development locally obviously)
* If you really want to trigger the message and send a test you will need to force the job to get queued.
    * Keep in mind it will also depend on what you have in your dev DB.
    * You will first need to set the `audit_enabled` settings value to `true` in your local or test yml file
    * Then you will need to queue up the job somehow
         * One way to do this is just put it as the first line in a controller method you then hit
         * Make sure to include the file `require 'claims_api/report_hourly_unsuccessful_submissions'`
         * Adding `ClaimsApi::ReportHourlyUnsuccessfulSubmissions.perform_async` should be all that is needed to trigger it to queue up
         * I would recommend stopping sidekiq first so you can see it actually queued up.

## Screenshots
Screenshot 2024-02-23 at 8 12 09 AM

## What areas of the site does it impact?
	modified:   lib/periodic_jobs.rb
	new file:   modules/claims_api/app/services/claims_api/slack/failed_submissions_messenger.rb
	new file:   modules/claims_api/app/sidekiq/claims_api/report_hourly_unsuccessful_submissions.rb
	new file:   modules/claims_api/spec/services/failed_submissions_messenger_spec.rb
	new file:   modules/claims_api/spec/sidekiq/report_hourly_unsuccessful_submissions_spec.rb

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by rjohnson2011 — DISMISSED** [review]

File: modules/claims_api/app/sidekiq/claims_api/report_hourly_unsuccessful_submissions.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    private
+
+    def errored_submissions_exist?
+      [@errored_claims, @errored_poa, @errored_itf, @errored_ews].any? { |var| var.count.positive? }
+    end
+
+    def skip_processing?
+      !Rails.env.production?
+    end
```
Comment (FonzMP) [inline modules/claims_api/app/sidekiq/claims_api/report_hourly_unsuccessful_submissions.rb:53]: I think your goal here is to only set this in the "production" API but this stands true in all environments. We'll need to trigger this off of a settings value. 
For example, staging
```
irb(main):001:0> Rails.env
=> "production"
```

ex: https://github.com/department-of-veterans-affairs/vets-api/blob/e66ef88d7c87d445ce5471bbec310cac005967dc/modules/claims_api/app/sidekiq/claims_api/claim_auditor.rb#L6

File: modules/claims_api/app/services/claims_api/slack/failed_submissions_messenger.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      end
+
+      def build_heading_message
+        heading_message = ''.dup
+        heading_title = "*ERRORED SUBMISSIONS* \n\n"
+        heading_message << heading_title
+        heading_timeframe = "#{@from} - #{@to} \n"
+        heading_message << heading_timeframe
+        heading_sub_text = "The following submissions have encountered errors in *#{@environment}*. \n\n"
+        heading_message << heading_sub_text
```
Comment (FonzMP) [inline modules/claims_api/app/services/claims_api/slack/failed_submissions_messenger.rb:54]: do we need the duplication in each of these? what's ur thoughts on combining each of these lines? 
```
heading_message << "*ERRORED SUBMISSIONS* \n\n"
heading_message << "#{@from} - #{@to} \n"
heading_message << "The following submissions have encountered errors in *#{@environment}*. \n\n"

File: modules/claims_api/app/services/claims_api/slack/failed_submissions_messenger.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        heading
+      end
+
+      def build_heading_message
+        heading_message = ''.dup
+        heading_title = "*ERRORED SUBMISSIONS* \n\n"
+        heading_message << heading_title
+        heading_timeframe = "#{@from} - #{@to} \n"
+        heading_message << heading_timeframe
+        heading_sub_text = "The following submissions have encountered errors in *#{@environment}*. \n\n"
```
Comment (FonzMP) [inline modules/claims_api/app/services/claims_api/slack/failed_submissions_messenger.rb:53]: where is `@environment`?

File: modules/claims_api/app/sidekiq/claims_api/report_hourly_unsuccessful_submissions.rb (hunk 1 above)
Comment (FonzMP) [reply modules/claims_api/app/sidekiq/claims_api/report_hourly_unsuccessful_submissions.rb]: sorry, the intent here was to use the existing value as it's only set in production. We could create a new settings value like you have in the new changes, but I'd probably just roll with the existing value unless you have a specific reason to create a new one. 

https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/2621bf3b0934b07b7e33947c4053147ede660004/apps/vets-api/prod/values.yaml#L2294

Comment (rockwellwindsor-va) [reply modules/claims_api/app/sidekiq/claims_api/report_hourly_unsuccessful_submissions.rb]: @FonzMP This makes sense, updated to utilize that value.

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**